commit fb0f3d281b7f81a11e210783940f3798c4744179
Author: Anup Patel <apatel@ventanamicro.com>
Date:   Tue Mar 28 09:22:20 2023 +0530

    RISC-V: Allow marking IPIs as suitable for remote FENCEs
    
    To do remote FENCEs (i.e. remote TLB flushes) using IPI calls on the
    RISC-V kernel, we need hardware mechanism to directly inject IPI from
    the supervisor mode (i.e. RISC-V kernel) instead of using SBI calls.
    
    The upcoming AIA IMSIC devices allow direct IPI injection from the
    supervisor mode (i.e. RISC-V kernel). To support this, we extend the
    riscv_ipi_set_virq_range() function so that IPI provider (i.e. irqchip
    drivers can mark IPIs as suitable for remote FENCEs.
    
    Signed-off-by: Anup Patel <apatel@ventanamicro.com>
    Reviewed-by: Atish Patra <atishp@rivosinc.com>
    Acked-by: Palmer Dabbelt <palmer@rivosinc.com>
    Signed-off-by: Marc Zyngier <maz@kernel.org>
    Link: https://lore.kernel.org/r/20230328035223.1480939-5-apatel@ventanamicro.com

diff --git a/arch/riscv/kernel/sbi-ipi.c b/arch/riscv/kernel/sbi-ipi.c
index 41981ab88493..a4559695ce62 100644
--- a/arch/riscv/kernel/sbi-ipi.c
+++ b/arch/riscv/kernel/sbi-ipi.c
@@ -72,6 +72,6 @@ void __init sbi_ipi_init(void)
 			  "irqchip/sbi-ipi:starting",
 			  sbi_ipi_starting_cpu, NULL);
 
-	riscv_ipi_set_virq_range(virq, BITS_PER_BYTE);
+	riscv_ipi_set_virq_range(virq, BITS_PER_BYTE, false);
 	pr_info("providing IPIs using SBI IPI extension\n");
 }